Makefile

```diff
@@ -0,0 +1,81 @@
+help:
+	@echo "check - clean the environment, install, lint, and run tests"
+	@echo "clean - remove Python file artifacts"
+	@echo "clean-all - remove *all* build artifacts"
+	@echo "clean-env - remove virtual environment"
+	@echo "deps - install the dependencies"
+	@echo "env - set up the virtual environment"
+	@echo "lint - check Python lint"
+	@echo "test - run tests"
+	@echo "test-deps - install the dependencies needed for running tests"
+
+###########
+# VARIABLES
+###########
+ENV := env/
+BIN := $(ENV)bin/
+FLAKE8 := $(BIN)flake8
+NOSETESTS := $(BIN)nosetests
+PIP := $(BIN)pip
+PYTHON := $(BIN)python
+
+#######
+# SETUP
+#######
+$(PYTHON):
+	virtualenv $(ENV)
+
+$(PIP):
+	virtualenv $(ENV)
+
+.PHONY: env
+env: $(PYTHON)
+
+##############
+# DEPENDENCIES
+##############
+.PHONY: test-deps
+test-deps: $(PIP)
+	$(PIP) install -r test-requirements.txt
+
+.PHONY: deps
+deps: $(PIP)
+	$(PIP) install -r requirements.txt
+
+# Nose is our test requirements canary
+$(NOSETESTS): $(PYTHON) deps
```

Comment:
Regarding @mitechie's comment earlier, since deps is a phony (not a real target on the file system with a timestamp), running these tests will cause the deps target to fire every time.

So deps and test-deps should probably have a canary file.  I'll add comments above so it'll make more sense.